### PR TITLE
Stats: Extend UTM metrics query for the specific post details page

### PR DESCRIPTION
**Pull request by dognose24**
Related to #87567 

## Proposed Changes

* Fix the returned key of the `useIsAdvancedFeatureEnabled` hook.
* Extend the `useUTMMetricsQuery` hook to support querying metrics for a specific post.

## Testing Instructions

* Spin this change with the Calypso Live link.
* Navigate to the Stats > Traffic page with the feature flag `stats/utm-module`.
* Ensure the UTM module works as previously.

截圖 2024-03-01 上午1 17 22

* Navigate to a specific post details page with the feature flag `stats/utm-module`: `/stats/post/{post-id}/{site-slug}?flags=stats/utm-module`.
* Ensure the UTM module on the post details page shows metrics for the post.

截圖 2024-03-01 上午1 17 40

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/state/stats/utm-metrics/selectors.ts [hunk 1]
```diff
@@ -10,8 +10,16 @@ const EMPTY_RESULT = [] as Array< UTMMetricItem >;
  * @param   {number}  siteId   Site ID
  * @returns {Object}           Highlights object; see schema.
  */
-export function getMetrics( state: object, siteId: number ) {
-	return get( state, [ 'stats', 'utmMetrics', 'data', siteId, 'metrics' ], EMPTY_RESULT );
+export function getMetrics( state: object, siteId: number, postId: number | undefined ) {
```
Comment (kangzj) [inline client/state/stats/utm-metrics/selectors.ts]: Should we set a default value to postId to identify the param is optional? Same applies for the hook and couple of places too.

**Review by kangzj — APPROVED** [review]
Works well for me and looks good! 👍

File: client/state/stats/utm-metrics/selectors.ts (hunk 1 above)
Comment (dognose24) [reply client/state/stats/utm-metrics/selectors.ts]: Do you mean something like https://github.com/Automattic/wp-calypso/pull/88042/commits/1b5114e9a93f0010ca9bc56b1cb0ac113270e4f6?

**Review by grzegorz-cp — APPROVED** [review]
👍

File: client/state/stats/utm-metrics/selectors.ts (hunk 1 above)
Comment (kangzj) [reply client/state/stats/utm-metrics/selectors.ts]: I like it! 👍

**Pull request merged**